packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/exceptions/ClientResetRequiredException.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.mongodb.exceptions
+
+import io.realm.kotlin.internal.interop.RealmAppPointer
+import io.realm.kotlin.internal.interop.RealmInterop
+import io.realm.kotlin.internal.interop.sync.SyncError
+
+/**
+ * Class encapsulating information needed for handling a Client Reset event.
+ *
+ * **See:** [Client Reset](https://www.mongodb.com/docs/atlas/app-services/sync/error-handling/client-resets/)
+ * for more information about when and why Client Reset occurs and how to deal with it.
+ *
+ */
+public class ClientResetRequiredException constructor(
+    private val appPointer: RealmAppPointer,
+    private val error: SyncError
+) : Throwable(message = error.detailedMessage) {
```

Comment:
IMO, we should format this error the same way we do in `RealmSyncUtils.createMessageFromSyncError()`. Just using the `detailedMessage` risk losing some valuable information, like the exact error code from the server.